### ISSUE TITLE
Server only tag

### ISSUE THREAD

**Issue by sarahdayan** [gridsome/gridsome#1056]
## Summary

There are times when you only want to compile at build time and not have that code run again in the client. For example, I'm building a navigation from a bunch of sources that I'm loading with `@gridsome/source-filesystem` in my `gridsome.config.js`. Then, in one on my layouts, I load them all with a GraphQL query and I build the navigation within a computed property.

This operation has a cost (I optimized it as much as possible but building a deeply nested hierarchy is inherently costly) and I'm using recent JavaScript features (e.g., `Array.prototype.flat`) which isn't available everywhere.

Since I don't need to rebuild the navigation at runtime, it would be really helpful to be able to only run some code when building, and rely on the memoized result. This would be the counterpart of `<ClientOnly>`.

### Basic example

```vue
<template>
  <h2>{{ WillPotentiallyReRenderAtRuntime }}</h2>
  <ServerOnly>
    {{ DoesNotNeedToReRenderEver }}
  </ServerOnly>
</template>

export default {
  computed: {
    WillPotentiallyReRenderAtRuntime, // regular computed property
    DoesNotNeedToReRenderEver, // only run at build time, then memoized forever
  }
}
```

The final code could code split `DoesNotNeedToReRenderEver`, making the build even lighter.

### Motivation

Such a feature would allow developers to fully defer expensive tasks to the server when they don't need to ever happen on the client. This would also let them add fewer dependencies because they wouldn't have to cater for older browsers in order to support the used features.

For example, syntax highlighting could be done entirely at build time, which would allow not to ship heavy libraries and tokenizing/lexing code snippets over and over on the client.

This feature is missing from almost every JavaScript static website generator and it would be awesome to see Gridsome be one the first to support it!

**Comment by hjvedvik** [comment]
Yes, we can consider having a builtin `<ServerOnly>` component since Gridsome already has a `<ClientOnly>` component. It is possible to do server only rendering with [`vue-lazy-hydration`](https://github.com/maoberlehner/vue-lazy-hydration), but that package could instead only be necessary for more advanced use cases.

**Comment by sarahdayan** [comment]
@hjvedvik Yes indeed! The beauty of a `<ServerOnly>` component is the "battery included" aspect, and how this would automatically code split code that's not needed at runtime.
